content/js/record.md

```diff
@@ -1,307 +1,331 @@
+Retrieve record using a Query object:
 
 ``` javascript
-const query = new skygear.Query(Note);
-query.equalTo('_id', '<your-note-_id>');
-
-skygear.publicDB.query(query)
-.then((records) => {
-  const note = records[0];
-  note['content'] = 'Hello New World';
-  return skygear.publicDB.save(note);
-}).then((record) => {
-  console.log('update success');
-}, (error) => {
-  console.error(error);
-});
+const blogQuery = new skygear.Query(Blog)
+  .greaterThan('popular', 10)
+  .addDescending('popular');
+
+blogQuery.limit = 10;
+
+skygear.publicDB.query(blogQuery)
+  .then((records) => {
+    console.log(records)
+  })
+  .catch((error) => {
+    console.error(error);
+  });
+
 ```
 
-- After saving a record, any attributes modified from the server side will
-be updated on the saved record object in place.
-- The local transient fields of the records are merged with any remote
-transient fields applied on the server side.
-- There is a shorter way for updating records, but only use it when you
-know what you are doing. (Unspecified field or column will not be changed,
-so in the case below only content field will be changed)
+### Updating records
+
+Update a record by ID:
 
 ``` javascript
-skygear.publicDB.save(new Note({
-  _id: 'note/<your-note-_id>',
+const noteUpdate = new Note({
+  _id: 'note/f705fd10-bfd8-47dd-aec9-3e3de32a8f5a',
   content: 'Hello New World'
-}));
-```
+});
+
+skygear.publicDB.save(noteUpdate)
+  .then((record) => {
+    console.log('Update Success!')
+  })
+  .catch((error) => {
+    console.error(error);
+  });
 
-### Deleting a record
 
-``` javascript
-skygear.publicDB.delete({
-  id: 'note/<your-note-_id>'
-}).then((record) => {
-  console.log(record);
-}, (error) => {
-  console.error(error);
-});
 ```
 
-You can also delete multiple records at one time.
+Updating multiple records:
+
 
 ``` javascript
-const query = new skygear.Query(Note);
-query.lessThan('rating', 3);
-
-const foundNotes = [];
-skygear.publicDB.query(query)
-.then((notes) => {
-  console.log(`Found ${notes.length} notes, going to delete them.`);
-  foundNotes = notes;
-  return skygear.publicDB.delete(notes); // return a Promise object
-})
-.then((errors) => {
-  errors.forEach((perError, idx) => {
-    if (perError) {
-      console.error('Fail to delete', foundNotes[idx]);
+const noteQuery = new skygear.Query(Note)
+  .equalTo('content', 'Hello World!');
+
+skygear.publicDB.query(noteQuery)
+  .then((records) => {
+    for(let i = 0; i < records.length; i++) {
+      records[i].content = 'Hello New World!';
     }
+    return skygear.publicDB.save(records);
   });
-}, (reqError) => {
-  console.error('Request error', reqError);
-});
+
 ```
 
-<a name="reference"></a>
-## Records Relations
+- After saving a record, any attributes modified from the server side will
+be updated on the saved record object in place.
+- The local transient fields of the records are merged with any remote
+transient fields applied on the server side.
```

Comment:
I think these should be placed before some code? at least before "updating multiple records"?